### ISSUE TITLE
Using PluginDescription for VPSet leads to validation failure if the vector contains different plugins

### ISSUE THREAD

**Issue by makortel** [cms-sw/cmssw#35426]
https://github.com/cms-sw/cmssw/pull/35385#issuecomment-927725483 shows that using a `PluginDescription` for vector of ParameterSets for different helper plugins along
```
initialStepTrajectoryFilterPreSplitting = cms.PSet(
    ComponentType = cms.string('CompositeTrajectoryFilter'),
    filters = cms.VPSet(
        cms.PSet( refToPSet_ = cms.string('initialStepTrajectoryFilterBasePreSplitting')),
        cms.PSet( refToPSet_ = cms.string('initialStepTrajectoryFilterShapePreSplitting'))),
)
```
leads to PSet validation to fail if the contained PSets point to different helper plugins.

**Comment by makortel** [comment]
assign core

**Comment by makortel** [comment]
I think this happens because 
* `addVPSet()` takes one `ParameterSetDescription` as a common validation template https://github.com/cms-sw/cmssw/blob/c1cb2f142a0d5a0f70961eb8102da9ff87d6b1a5/FWCore/ParameterSet/interface/ParameterSetDescription.h#L148-L151
* VPSet validation uses the same `ParameterSetDescription` object to validate each element https://github.com/cms-sw/cmssw/blob/c1cb2f142a0d5a0f70961eb8102da9ff87d6b1a5/FWCore/ParameterSet/src/ParameterDescription.cc#L226-L228
* `PluginDescription` caches the plugin https://github.com/cms-sw/cmssw/blob/c1cb2f142a0d5a0f70961eb8102da9ff87d6b1a5/FWCore/ParameterSet/interface/PluginDescription.h#L125-L127 https://github.com/cms-sw/cmssw/blob/c1cb2f142a0d5a0f70961eb8102da9ff87d6b1a5/FWCore/ParameterSet/interface/PluginDescription.h#L219-L223

**Comment by wddgit** [comment]
We discussed this at the Core meeting today.

Here is the documentation for configuration validation with plugin helpers:

https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#A_Plugin_Module_Using_Another_He

Here are a couple initial comments.

1. The current design does not support putting helper plugin ParameterSets into VPSet's. This is not a bug. This is the original design Chris and I agreed to when I implemented this in 2018. I am not saying we cannot add support for this, just that the current code does not support this. That was not the original design.

2. It is possible with the current design to support multiple helper plugins in one module. It requires using different names at the proper points. You need to use multiple separate PSets instead of a VPSet. I'm not sure if that I good enough to support the current use case we are discussing, but that should work now without any changes to the Framework.

I am starting to look at how one might implement this with a VPSet. If this isn't needed and using multiple separate PSets is sufficient, please let me know and I'll go back to working on concurrent runs.

**Comment by wddgit** [comment]
After looking at this more closely I realized the previous comment wasn't correct and should be disregarded. The current version does support VPSets but with the restriction that all elements have the same plugin type. I submitted PR #35477 to remove that restriction.

Other kinds of VPSets must have all elements passing validation with the same ParameterSetDescription, but in this special case it is not hard to avoid this restriction.
